soh/soh/Enhancements/randomizer/location_access/dungeons/fire_temple.cpp

```diff
@@ -626,114 +810,227 @@ void RegionTable_Init_FireTemple() {
+        Entrance(RR_FIRE_TEMPLE_MQ_HIGH_TORCH_ROOM,      []{return false;}),
+        Entrance(RR_FIRE_TEMPLE_MQ_HIGH_TORCH_ROOM_CAGE, []{return false;}),
+        Entrance(RR_FIRE_TEMPLE_MQ_BIG_LAVA_ROOM,        []{return logic->TakeDamage();}),
     });
 
-    areaTable[RR_FIRE_TEMPLE_MQ_HIGH_TORCH_ROOM] = Region("Fire Temple MQ High Torch Room", SCENE_FIRE_TEMPLE, {}, {
+    areaTable[RR_FIRE_TEMPLE_MQ_HIGH_TORCH_ROOM] = Region("Fire Temple MQ High Torch Room", SCENE_FIRE_TEMPLE, {
+        //Events
+        EventAccess(LOGIC_FIRE_MQ_HIGH_TORCH_LIT, []{return (logic->CanUse(RG_FIRE_ARROWS) && logic->FireTimer() >= 24);}),
+    }, {
         //Locations
-        LOCATION(RC_FIRE_TEMPLE_MQ_FLAME_WALL_POT_1,            logic->CanBreakPots()),
-        LOCATION(RC_FIRE_TEMPLE_MQ_FLAME_WALL_POT_2,            logic->CanBreakPots()),
-        LOCATION(RC_FIRE_TEMPLE_MQ_LAVA_TORCH_CRATE_1,          logic->CanBreakCrates()),
-        LOCATION(RC_FIRE_TEMPLE_MQ_LAVA_TORCH_CRATE_2,          logic->CanBreakCrates()),
-        LOCATION(RC_FIRE_TEMPLE_MQ_LAVA_TORCH_CRATE_3,          logic->CanBreakCrates()),
-        LOCATION(RC_FIRE_TEMPLE_MQ_LAVA_TORCH_CRATE_4,          logic->CanBreakCrates()),
-        LOCATION(RC_FIRE_TEMPLE_MQ_LAVA_TORCH_CRATE_5,          logic->CanBreakCrates()),
-        LOCATION(RC_FIRE_TEMPLE_MQ_LAVA_TORCH_SMALL_CRATE_1,    logic->CanBreakSmallCrates()),
-        LOCATION(RC_FIRE_TEMPLE_MQ_LAVA_TORCH_SMALL_CRATE_2,    logic->CanBreakSmallCrates()),
-        LOCATION(RC_FIRE_TEMPLE_MQ_LAVA_TORCH_SMALL_CRATE_3,    logic->CanBreakSmallCrates()),
-        LOCATION(RC_FIRE_TEMPLE_MQ_LAVA_TORCH_SMALL_CRATE_4,    logic->CanBreakSmallCrates()),
-        LOCATION(RC_FIRE_TEMPLE_MQ_LAVA_TORCH_SMALL_CRATE_5,    logic->CanBreakSmallCrates()),
+        LOCATION(RC_FIRE_TEMPLE_MQ_FLAME_WALL_POT_1,         logic->CanBreakPots() && logic->FireTimer() >= 24),
+        LOCATION(RC_FIRE_TEMPLE_MQ_FLAME_WALL_POT_2,         logic->CanBreakPots() && logic->FireTimer() >= 24),
+        LOCATION(RC_FIRE_TEMPLE_MQ_LAVA_TORCH_CRATE_1,       logic->CanBreakCrates() && logic->FireTimer() >= 24),
+        LOCATION(RC_FIRE_TEMPLE_MQ_LAVA_TORCH_CRATE_3,       logic->CanBreakCrates() && logic->FireTimer() >= 24),
+        LOCATION(RC_FIRE_TEMPLE_MQ_LAVA_TORCH_CRATE_4,       logic->CanBreakCrates() && logic->FireTimer() >= 16),
+        LOCATION(RC_FIRE_TEMPLE_MQ_LAVA_TORCH_SMALL_CRATE_2, logic->CanBreakSmallCrates() && logic->FireTimer() >= 24),
+        LOCATION(RC_FIRE_TEMPLE_MQ_LAVA_TORCH_SMALL_CRATE_3, logic->CanBreakSmallCrates() && logic->FireTimer() >= 16),
     }, {
         //Exits
-        Entrance(RR_FIRE_TEMPLE_MQ_UPPER_MAZE,       []{return logic->SmallKeys(SCENE_FIRE_TEMPLE, 3);}),
-        Entrance(RR_FIRE_TEMPLE_MQ_NARROW_PATH_ROOM, []{return true;}),
+        Entrance(RR_FIRE_TEMPLE_MQ_HIGH_TORCH_ROOM_ABOVE_CAGE, []{return logic->IsAdult || logic->CanUse(RG_HOOKSHOT) && logic->FireTimer() >= 24;}),
```

Comment:
```suggestion
        Entrance(RR_FIRE_TEMPLE_MQ_HIGH_TORCH_ROOM_ABOVE_CAGE, []{return (logic->IsAdult || logic->CanUse(RG_HOOKSHOT)) && logic->FireTimer() >= 24;}),
```
&& binds more tightly than ||